src/lib.rs

```diff
@@ -0,0 +1,291 @@
+use std::collections::HashSet;
+use std::fmt;
+
+use guest::prelude::*;
+use kubewarden_policy_sdk::wapc_guest as guest;
+
+use k8s_openapi::api::core::v1::{EnvVar, PodSpec};
+extern crate kubewarden_policy_sdk as kubewarden;
+use encoding::all::ASCII;
+use encoding::{DecoderTrap, Encoding};
+use kubewarden::{protocol_version_guest, request::ValidationRequest, validate_settings};
+use rusty_hog_scanner::{SecretScanner, SecretScannerBuilder};
+use std::string::String;
+
+mod settings;
+use settings::Settings;
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+/// Represents a secret that has been found in an env var
+#[derive(Eq, Hash, PartialEq, Debug)]
+struct SecretRejected {
+    /// nome of the container where the secret was found
+    container: String,
+    /// reason of rejection. It describes the secret that it was found
+    reason: String,
+    /// key of the env var
+    key: String,
+}
+
+impl fmt::Display for SecretRejected {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "container: {}, key: {}, reason: {}. ",
+            self.container, self.key, self.reason
+        )
+    }
+}
+
+fn validate(payload: &[u8]) -> CallResult {
+    let validation_request: ValidationRequest<Settings> = ValidationRequest::new(payload)?;
+
+    match validation_request.extract_pod_spec_from_object() {
+        Ok(pod_spec) => {
+            if let Some(pod_spec) = pod_spec {
+                return validate_pod_spec(pod_spec);
+            }
+            // If there is not pod spec, just accept it. There is no data to be validated.
+            kubewarden::accept_request()
+        }
+        Err(_) => {
+            // We were forwarded a request we cannot unmarshal or
+            // understand, just accept it
+            kubewarden::accept_request()
```

Comment:
I would reject with and provide the error message into the body

Reply:
done https://github.com/raulcabello/env-variable-secrets-scanner-policy/commit/8b5996690f444e23bae9c5919c75d156373fb249 as this is what we do in other policies
However I'm not very convinced we should do this. What would happen if someone configure the policy rules resources to evaluate more resources than just `Pods` or high level objects that contain Pods? If a `*` is added, all resources except `Pods` would be rejected which would break the k8s cluster.